client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-add-license-notification/index.tsx

```diff
@@ -39,52 +40,107 @@ export default function SiteAddLicenseNotification() {
 		dispatch( setPurchasedLicense( license.selectedProducts.length ? license : undefined ) );
 	};
 
+	function getTranslatedLicenseAssignmentMessage(
+		licenses: Array< ProductInfo >,
+		type: 'fulfilled' | 'rejected'
+	) {
+		const licenseNames = licenses.map( ( license ) => license.name );
+		const initialLicenseList = licenseNames.slice( 0, -1 );
+		const lastLicenseItem = licenseNames.slice( -1 )[ 0 ];
+
+		const commaCharacter = translate( ',', {
+			comment:
+				'The character used to separate items in a list, such as the comma in "Backup, Scan, and Boost".',
+		} );
+		const conjunction =
+			licenses.length > 2
+				? translate( `%(commaCharacter)s and`, {
+						args: {
+							commaCharacter,
+							comment:
+								'The final separator of a delimited list, such as ", and" in "Backup, Scan, and Boost."',
+						},
+				  } )
+				: translate( ' and', {
+						args: {
+							comment:
+								'The way that two words are separated, such as " and" in "Backup and Scan". Note that the space here is important due to the way the final string is constructed.',
+						},
+				  } );
+		const multipleLicensesArgs = {
+			selectedSite,
+			initialLicenseList: initialLicenseList.join( `${ commaCharacter } ` ),
+			lastLicenseItem: lastLicenseItem,
+			conjunction,
+		};
+		const singleLicenseArgs = {
+			selectedSite,
+			licenseItem: lastLicenseItem,
+		};
+		const components = {
+			strong: <strong />,
+			em: <em />,
+		};
+
+		if ( type === 'fulfilled' ) {
+			// We are not using the same translate method for plural form since we have different arguments.
+			return licenses.length > 1
+				? translate(
+						'{{strong}}%(initialLicenseList)s%(conjunction)s %(lastLicenseItem)s{{/strong}} ' +
```

Comment:
In some languages the word 'and' is connected to the following word, so it could be better to leave no space between `%(conjunction)s` and `%(lastLicenseItem)s` and instead set the original to ` and ` and let the translations remove spaces as necessary.

Reply:
Thanks, I have made the changes.